PSReadLine/Prediction.Views.cs

```diff
@@ -0,0 +1,780 @@
+            {
+                bool inputUnchanged = string.Equals(_inputText, userInput, _singleton._options.HistoryStringComparison);
+                if (!inputUnchanged && _selectedIndex > -1)
+                {
+                    // User input was changed while a specific list item was selected. This is a strong indicator that
+                    // the selected list item was accepted, because the user was editing based on the selected item.
+                    //
+                    // But be noted that it's not guaranteed to be 100% accurate, because it's still possible that the
+                    // user selected all the buffer and then was replacing it with something totally different.
+                    OnSuggestionAccepted();
+                }
+
+                if (!WindowSizeMeetsMinRequirement)
+                {
+                    // If the window size is too small for the list view to work, we just disable the list view.
+                    Reset();
+                    return;
+                }
+
+                _inputText = userInput;
+                _selectedIndex = -1;
+                _listItemWidth = Math.Min(_singleton._console.BufferWidth, ListMaxWidth);
+
+                if (inputUnchanged)
+                {
+                    // This could happen when the user types 'ctrl+z' (undo) while looping through the suggestion list.
+                    // The 'undo' operation would revert the line back to the original user input, and in that cases, we
+                    // could reuse the existing suggestion results.
+                    return;
+                }
+
+                _listItems?.Clear();
+
+                try
+                {
+                    if (UsePlugin)
+                    {
+                        PredictInput();
+                    }
+
+                    if (UseHistory)
+                    {
+                        _listItems = GetHistorySuggestions(userInput, ListMaxCount);
+                    }
+                }
+                catch
+                {
+                    Reset();
+                }
+            }
+
+            /// <summary>
+            /// Aggregate the suggestion results from both the prediction API and the history.
+            /// </summary>
+            /// <remarks>
+            /// If the prediction source contains both the history and plugin, then we allocate 3
+            /// slots at most for history suggestions. The remaining slots are evenly distributed
+            /// to each predictor plugin.
+            /// </remarks>
+            private void AggregateSuggestions()
+            {
+                var results = GetPredictionResults();
+                if (results?.Count > 0)
+                {
+                    try
+                    {
+                        _listItems ??= new List<SuggestionEntry>();
+                        _cacheList1 ??= new List<int>();
```

Comment:
Is there something more descriptive than cachelist1 and 2?  or have a comment describing how these are used if they are two caches of the same thing.

Reply:
I cannot find a suitable descriptive name for those 2 cache lists, but I will add comments here to explain what they are for.